packages/core/utils/src/modules-sdk/types/medusa-service.ts

```diff
@@ -0,0 +1,246 @@
+import { DmlEntity } from "../../dml"
+import {
+  Constructor,
+  Context,
+  FindConfig,
+  Pluralize,
+  RestoreReturn,
+  SoftDeleteReturn,
+} from "@medusajs/types"
+
+export type BaseMethods =
+  | "retrieve"
+  | "list"
+  | "listAndCount"
+  | "delete"
+  | "softDelete"
+  | "restore"
+  | "create"
+  | "update"
+
+export type ModelDTOConfig = {
+  dto: object
+  dml?: DmlEntity<any, any>
```

Comment:
nitpicky/suggestion: Can we call this `model`? I am afraid of using the `DML` in the public-facing API in case we change it to be a "product" of its own if that makes sense. DML could evolve to be the name of an implementation detail. Wdyt?
```suggestion
  model?: DmlEntity<any, any>
```

Reply:
Maybe to add some context to this: If we choose to productise the DML, it will likely have a more "commercial" name, similar to how Drizzle is named Drizzle, Prisma is named Prisma, etc. This is what I mean when I say the DML term could become purely an implementation detail. 

Therefore, if we can establish a public-facing API, that won't have to change if we choose to do this, I think that would be nice. Less breaking changes to worry about 😄

Reply:
I agree with this but for a different reason: we already established data models as `model` so it would be easier to understand for developers.

Reply:
I think we can go with that and eventually settle definately on it later, I ll include those changes

Reply:
yes for sure, we might also need to cleanup what is called entity to model, but we can do that in second time, at least we dont add more of them